code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubUtils.java

```diff
@@ -0,0 +1,97 @@
+package io.cattle.platform.iaas.api.auth.github;
+
+import io.cattle.platform.token.TokenException;
+import io.cattle.platform.token.TokenService;
+import io.cattle.platform.util.type.CollectionUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class GithubUtils {
+
+    private TokenService tokenService;
+
+    public String validateAndFetchGithubToken(String token) {
+        Map<String, Object> jsonData = getJsonData(token);
+        if(null == jsonData) {
+            return null;
+        }
+        return ObjectUtils.toString(jsonData.get("access_token"), null);
+    }
+    
+    public String validateAndFetchAccountIdFromToken(String token) {
+        Map<String, Object> jsonData = getJsonData(token);
+        if(null == jsonData) {
+            return null;
+        }
+        return ObjectUtils.toString(jsonData.get("account_id"), null);
+    }
+    
+    @SuppressWarnings("unchecked")
+    public List<String> validateAndFetchOrgIdsFromToken(String token) {
+        Map<String, Object> jsonData = getJsonData(token);
+        if(null == jsonData) {
+            return null;
+        }
+        return (List<String>) CollectionUtils.toList(jsonData.get("org_ids"));
+    }
+    
+    @SuppressWarnings("unchecked")
+    public List<String> validateAndFetchTeamIdsFromToken(String token) {
+        Map<String, Object> jsonData = getJsonData(token);
+        if(null == jsonData) {
+            return null;
+        }
+        return (List<String>) CollectionUtils.toList(jsonData.get("team_ids"));
+    }
+    
+    @SuppressWarnings("unchecked")
+    public Pair<String, Pair<List<String>, List<String>>> validateAndFetchAccesibleIdsFromToken(String token) {
```

Comment:
For future reference, if you find yourself needing to use Pair objects it's a good indication that you should have a proper data structure.  Even a plain class with public fields and no accessors is better than a Pair object.